design/use-ironic.md

```diff
@@ -0,0 +1,151 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# use-ironic
+
+## Status
+
+One of: implementable
+
+## Table of Contents
+
+<!--ts-->
+   * [use-ironic](#use-ironic)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-18T16:27-05:00 -->
+
+<!--te-->
+
+## Summary
+
+Part of being a bare metal provisioning system is having the ability
+to control and provision images to bare metal hosts. Many such systems
+exist, so rather than build a new one we want to select an existing on
+that supports our needs. Initially this will be
+[Ironic](https://docs.openstack.org/ironic/latest/).
+
+## Motivation
+
+We want to choose a tool that has an API to drive, provides good
+community support, can handle a variety of hardware, and can run in a
+Kubernetes environment. Ironic meets all of those criteria.
+
+### Goals
+
+1. Choose an existing provisioning/deployment tool to be used to
+   configure hosts and their images.
+2. Create an architecture that allows us to choose a different tool
+   later, if we have some reason to do that.
+
+### Non-Goals
+
+1. Build a new provisioning/deployment tool for the sake of having one
+   written in golang.
+2. Integrate with every provisioning/deployment tool potential users
+   of MetalKube might have in their data centers.
+
+## Proposal
+
+Package Ironic as an image in a way that allows us to drive it from an
+operator, but hide its use from the end user.
+
+### User Stories [optional]
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+- We want to eliminate the need for a MySQL database or RabbitMQ
+  message bus.
+- We need to include Ironic Inspector as well as the main Ironic API
+  so we can perform hardware introspection.
+- Both services should run inside the same Pod as the
+  baremetal-operator.
```

Comment:
Maybe expand on this a bit to explain why that is desirable?

Reply:
Done

Reply:
hm, don't see it?  To be clear, I was thinking of explaining somewhere why running within the same pod is a desirable thing.

I do like the additions that explain the MySQL and Rabbit changes!

Reply:
Nevermind, I see the security note later in the doc.